haystack/core/component/component.py

```diff
@@ -163,10 +163,12 @@ def run(self, **kwargs):
     else:
         run: Callable
 
+    supports_async: bool
```

Comment:
I think we shouldn't have this as part of the Component Protocol, I would also make it a dunder field like `__haystack_input__` and `__haystack_output__` and treat it in the same way.

Making it part of the protocol implies the user must declare it even though we shouldn't expect the user to do that. Also the user should have no need to access that information about a Component, that's why it should be a dunder.

Reply:
Fair point. The only place where I can foresee the need for this property is inside the pipeline class, so making it a dunder field should be fine. Will fix.